File: spring-observation/src/main/java/com/github/jrhee17/HelloController.java
package com.github.jrhee17;

import java.time.Duration;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.reactive.function.client.WebClient;

import io.micrometer.observation.ObservationRegistry;
import reactor.core.observability.micrometer.Micrometer;
import reactor.core.publisher.Mono;

/**
 * An example of a controller which uses {@link WebClient} inside.
 */
@RestController
public class HelloController {

    private final ObservationRegistry registry;

    HelloController(ObservationRegistry registry) {
        this.registry = registry;
    }

    @GetMapping("/hello")
    Mono<String> hello() {
        return Mono
                .just("world")
                .delayElement(Duration.ofMillis(1000));	//3
    }
}

File: spring-boot/src/main/kotlin/com/github/jrhee17/HelloController.kt
package com.github.jrhee17

import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController

@RestController
open class HelloController {

    @RequestMapping(method = [RequestMethod.GET], path = ["/"])
    fun index(): String {
        return "index"
    }
}